edx_rest_api_client/tests/test_client.py

```diff
@@ -70,3 +76,49 @@ def test_supplied_jwt(self):
         with mock.patch('edx_rest_api_client.auth.SuppliedJwtAuth.__init__', return_value=None) as mock_auth:
             EdxRestApiClient(URL, jwt=JWT)
             mock_auth.assert_called_with(JWT)
+
+
+@httpretty.activate
+@ddt.ddt
+class ClientCredentialTests(TestCase, AuthenticationTestMixin):
+    """ Test client credentials requests. """
+
+    URL = "http://test-auth/access_token/"
+
+    def test_get_client_credential_access_token_success(self):
+        """ Test that the get access token method handles 200 responses and returns the access token. """
+        code = 200
+        body = {"access_token": "my-token", "expires_in": 1000}
+        now = datetime.datetime.now(UTC)
+
+        expected_return = ("my-token", now + datetime.timedelta(seconds=1000))
+
+        with freeze_time(now):
+            self._mock_auth_api(URL, code, body=body)
+            self.assertEqual(
+                EdxRestApiClient.get_oauth_access_token(URL, "client_id", "client_secret"),
+                expected_return
+            )
+
+    @ddt.data(
+        (400, {"error": "denied"}, (None, None)),
+        (500, None, (None, None))
+    )
+    @ddt.unpack
+    def test_get_client_credential_access_token_failure(self, code, body, expected_return):
+        """ Test that the get access token method handles failure responses. """
+        self._mock_auth_api(URL, code, body=body)
+        self.assertEqual(
+            EdxRestApiClient.get_oauth_access_token(URL, "client_id", "client_secret"),
+            expected_return
+        )
+
+    def test_get_client_credential_access_token_no_url(self):
+        """ Test that the get access token method raises MissingSchema error when no url provided. """
+        with self.assertRaises(requests.exceptions.MissingSchema):
+            EdxRestApiClient.get_oauth_access_token(None, "client_id", "client_secret")
+
+    def test_get_client_credential_access_token_bad_url(self):
```

Comment:
Do we really need this test case? It's verifying some pretty generic behavior of the `requests` library.